### ISSUE TITLE
Possible to reload background image?

### ISSUE THREAD

**Issue by se7en-x230** [hyprwm/hyprlock#290]
Image: is reloading but Background does not have this ability?

**Comment by fjueic** [comment]
@se7en-x230 i used image and got this result
downside: no blur noise and other background only features
the interval was set to 1 sec making it a bit laggy

[1714749440.webm](https://github.com/hyprwm/hyprlock/assets/89282645/188f738e-7452-43fc-a6c7-3b27f2f9d929)

**Comment by mahaveergurjar** [comment]
How to due this @fjueic

**Comment by mahaveergurjar** [comment]
I need this for my music hyprlock layout

**Comment by mahaveergurjar** [comment]
![240511_12h48m10s_screenshot](https://github.com/hyprwm/hyprlock/assets/57872034/b03b1d22-a572-417b-aff9-8313e254b71c)
![240511_12h48m27s_screenshot](https://github.com/hyprwm/hyprlock/assets/57872034/4ef99327-e969-4f1e-b8bc-401cf85b178b)
![240511_12h48m38s_screenshot](https://github.com/hyprwm/hyprlock/assets/57872034/90410a01-288a-46f6-85c5-e7fe9aa2295f)

**Comment by mahaveergurjar** [comment]
In second picture you see the background not change ACC. to album background

**Comment by fjueic** [comment]
its in wiki
[here](https://wiki.hyprland.org/Hypr-Ecosystem/hyprlock/#image)
reload time and reload cmd

never tried it:-
reload time 0 if you want to handle it yourself instead of reloading every certain time interval

`pkill -USR1 hyprlock` to reload

**Comment by fjueic** [comment]
as for changing background image
i dont think you can 

in my example 
> i used image and got this result

i did not used background i just used a image and make it cover entire screen
but that way you will lose blur

**Comment by mahaveergurjar** [comment]
Okay but any way to reloading background because if image used then blur not achieve

**Comment by mahaveergurjar** [comment]
https://github.com/hyprwm/hyprlock/assets/57872034/98021ce5-d073-4bf8-916a-262adfd11c54

**Comment by mahaveergurjar** [comment]
how to set image ratio to set fit the image in 1920X1080
@fjueic

**Comment by mahaveergurjar** [comment]
I achieve blur but image ratio not acc. to 1920x1080

https://github.com/hyprwm/hyprlock/assets/57872034/a9ec8fdb-7e4e-41a8-8124-86b6f5ac82d3

**Comment by fjueic** [comment]
like this?
![Untitled-2024-02-03-2159](https://github.com/hyprwm/hyprlock/assets/89282645/28dfac8f-6f53-485e-a37e-b636e8066eb2)

**Comment by fjueic** [comment]
how did you achieve blur?

**Comment by mahaveergurjar** [comment]
i make script for this

**Comment by mahaveergurjar** [comment]
> like this? ![Untitled-2024-02-03-2159](https://private-user-images.githubusercontent.com/89282645/329798359-28dfac8f-6f53-485e-a37e-b636e8066eb2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU0NDc2NDMsIm5iZiI6MTcxNTQ0NzM0MywicGF0aCI6Ii84OTI4MjY0NS8zMjk3OTgzNTktMjhkZmFjOGYtNmY1My00ODVlLWEzN2UtYjYzNmU4MDY2ZWIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTExVDE3MDkwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRhYmZmNmRhNDRjYTY4NGQzZDlmMjE4OGI3N2UxNzdjODJlOGU0NjVjZjI0N2M2YzkwZWM4NzI3ZGY3OTMyNmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xLTMsfN_ftQNbZ4idd7ryTx9YdlY7siU5gMFdqdTTrc)

How to stretch the image to cover black area

**Comment by fjueic** [comment]
stretch? i don't like this idea
you can do this via command line tool and and store output in /tmp 
`convert input.png -resize 1920x1080\! output.png`

i don't like the idea personally

![image](https://github.com/hyprwm/hyprlock/assets/89282645/99518ba7-c882-4cdb-8b6a-b59b9e9e24b4)

**Comment by fjueic** [comment]
> i make script for this
> 
> ```
> #!/bin/env bash
> 
> THUMB=/tmp/hyde-mpris
> THUMB_BLURRED=/tmp/hyde-mpris-blurred
> 
> fetch_thumb() {
> artUrl=$(playerctl -p spotify metadata --format '{{mpris:artUrl}}') 
> [[ "${artUrl}" = "$(cat "${THUMB}.inf")" ]] && return 0
> 
> printf "%s\n" "$artUrl" > "${THUMB}.inf"
> 
> curl -so "${THUMB}.png" "$artUrl"
> convert "${THUMB}.png" -quality 50 "${THUMB}.png" && "${THUMB}.png"
>   # Create blurred version
>   convert "${THUMB}.png" -blur 20x20 "${THUMB_BLURRED}.png"
>   pkill -USR2 hyprlock
> }
> 
> { playerctl -p spotify metadata --format '{{title}}    {{artist}}'  && fetch_thumb ;} || { rm -f "${THUMB}*" && exit 1;}
> ```


ok pre-processing
i thought you did it natively

if you have a script stretching should be just one line away

**Comment by mahaveergurjar** [comment]
I try this and working thanks
![image](https://github.com/hyprwm/hyprlock/assets/57872034/6f3c55b6-ecfa-491a-8e24-bf1715716c6b)
but not see good any idea to see only zoom the center part of image to fill the screen

**Comment by fjueic** [comment]
`convert input.png -resize 1920x1080^ -gravity center -extent 1920x1080 output.png`
bruh say directly if there is more pre-processing you would like to do, don't keep adding 

![image](https://github.com/hyprwm/hyprlock/assets/89282645/c965b43f-0ae4-40db-a4b0-5504bbb18232)

**Comment by mahaveergurjar** [comment]
ok thanks

**Comment by sluedecke** [comment]
I hope this one is not forgotten ... I would love to be able to have "dynamic" images on my lock screen.

- `image.reload_cmd` does not work well since my image library is mainly jpeg and the reload accept png only.  I can write a conversion script, but then I don't want to convert stuff all the time
- normal `image` does not scale as nicely as a `background` image
- `background` has a nice blur feature and already scales the image used nicely
- I have already set up `hyprpapr` to use my collection of images for the background image

So I kindly request to have

`path = background`

as an option if it is possible.  TIA

**Comment by fjueic** [comment]
@sluedecke what part are you unable to do?

If you read the above comments, everything can be done. Blur and format can be pre processed

**Comment by sluedecke** [comment]
> If you read the above comments, everything can be done. Blur and format can be pre processed

You are right, I could script all of this.  But why should I, if all this application logic is already implemented in hyprlock?  From my perspective it seems to be more consistent if hyprlock supports other file types than PNG for image reloading and/or if it would support reloading for the background image.  Everything else (blurring, positioning, ...) is already there.

**Comment by fjueic** [comment]
> > If you read the above comments, everything can be done. Blur and format can be pre processed
> 
> You are right, I could script all of this. But why should I, if all this application logic is already implemented in hyprlock? From my perspective it seems to be more consistent if hyprlock supports other file types than PNG for image reloading and/or if it would support reloading for the background image. Everything else (blurring, positioning, ...) is already there.

fair point but I don't think anyone wants to implement it rn

[this](https://github.com/fjueic/.dotfiles/blob/main/.config/hypr/hyprlock.py) is my flow, image gets converted to png whenever i change the wallpaper variable. Note that i have hardcoded the monitor resolution.

as for blur and changing in intervals, it should be easy. Ask me if you can stuck
